lib/mongo/mongoClient.js

```diff
@@ -79,11 +79,8 @@ module.exports = function (config) {
         return clientCb(new Error('Client closing, cannot be used.'));
       }
 
-      mongoClient.db(databaseName).collection(collectionName, function (err, collection) {
-        if (err != null) {
-          clientCb(err);
-        }
-
+      var collection = mongoClient.db(databaseName).collection(collectionName);
+      process.nextTick(function () {
```

Comment:
Should this use `callbackify` for consistency?

Reply:
Not quite, the reasons for this are a bit different. callbackify is used to make an async function accept callbacks. In this case, the API for returning a collection no longer accepts a callback but is a regular function, not an async one. I'm just adding the process.nextTick so that previous expectations of callback behaviour is preserved.

Reply:
Can you please add a comment explaining why we need `nextTick`?

Reply:
Yes the idea is to prevent releasing "zalgo" as the term is called when a function that accepts a callback can return either immediately or sometimes with a delay, calling it after some calculation is done. This leads to unpredictable behaviour and subtle bugs. The old mongodb API had this delay, but I just realized process.nextTick is not needed here as this function always calls the callback immediately so I removed the process.nextTick.